plugin.video.greekvoice/resources/lib/runner.py

```diff
@@ -0,0 +1,187 @@
+
+
+@urldispatcher.register('directory', ['kind'])
+def directory(kind=None):
+
+    if kind == 'video':
+
+        menu = [
+            {
+                'title': ' '.join(['Greek Voice', __lang__(30005).format('1')]),
+                'icon': ICON,
+                'fanart': os.path.join(ART_FOLDER, 'greek_voice_fanart_1.jpg'),
+                'url': GREEK_VOICE_URL_1
+            }
+            ,
+            {
+                'title': ' '.join(['Greek Voice', __lang__(30005).format('2')]),
+                'icon': ICON,
+                'fanart': os.path.join(ART_FOLDER, 'greek_voice_fanart_2.jpg'),
+                'url': GREEK_VOICE_URL_2
+            }
+            ,
+            {
+                'title': ' '.join([__lang__(30004), __lang__(30005).format('1')]),
+                'icon': os.path.join(ART_FOLDER, 'tilemousiki.jpg'),
+                'fanart': os.path.join(ART_FOLDER, 'tilemousiki_fanart.jpg'),
+                'url': TILEMOUSIKI_URL_1
+            }
+            ,
+            {
+                'title': ' '.join([__lang__(30004), __lang__(30005).format('2')]),
+                'icon': os.path.join(ART_FOLDER, 'tilemousiki.jpg'),
+                'fanart': os.path.join(ART_FOLDER, 'tilemousiki_fanart.jpg'),
+                'url': TILEMOUSIKI_URL_2
+            }
+            ,
+            {
+                'title': ' '.join(['Greek Kids', __lang__(30005).format('1')]),
+                'icon': os.path.join(ART_FOLDER, 'greek_kids.jpg'),
+                'fanart': FANART,
+                'url': CARTOONS_URL_1
+            }
+            ,
+            {
+                'title': ' '.join(['Greek Kids', __lang__(30005).format('2')]),
+                'icon': os.path.join(ART_FOLDER, 'greek_kids.jpg'),
+                'fanart': FANART,
+                'url': CARTOONS_URL_2
+            }
+        ]
+
+    elif kind == 'audio':
+
+        menu = [
+            {
+                'title': 'WPSO Greek Voice Radio',
+                'icon': os.path.join(ART_FOLDER, 'wpso.jpg'),
+                'fanart': os.path.join(ART_FOLDER, 'radio_fanart.jpg'),
+                'url': WPSO_GREEK_VOICE_RADIO_URL
+            }
+            ,
+            {
+                'title': 'WXYB Radio GR IT ES',
+                'icon': os.path.join(ART_FOLDER, 'wxyb.jpg'),
+                'fanart': os.path.join(ART_FOLDER, 'radio_fanart.jpg'),
+                'url': WXYB_RADIO_GR_IT_ES
+            }
+            ,
+            {
+                'title': 'XAMOS Youth Radio',
+                'icon': os.path.join(ART_FOLDER, 'xamos.jpg'),
+                'fanart': os.path.join(ART_FOLDER, 'xamos_fanart.jpg'),
+                'url': XAMOS_URL
+            }
+        ]
+
+    elif kind == 'youtube':
+
+        menu = wrapper.list_channel_videos(channel_url=YT_CHANNEL_URL, limit=50)
+
+    else:
+
+        return
+
+    # noinspection PyUnboundLocalVariable
+    for m in menu:
+        list_item = xbmcgui.ListItem(m['title'])
```

Comment:
might consider to add offscreen=True as argument in the initializer (avoid gui locks)

Reply:
Done